### PR TITLE
`gw-update-posts.php`: Added support to update post date and time.

### PR DESCRIPTION
**Pull request by sbassah**
## Context

⛑️ Ticket(s): https://secure.helpscout.net/conversation/2128193021/42965?folderId=3775888

## Summary

This pull request adds support for updating the Post date and time using the Date and Time fields, respectively. The parameter to set the post date and time will be an array named date_time, with two keys, namely date and time, and the value set to the date and time field, respectively. 

new GW_Update_Posts( array(
    'form_id' => 123,
    'post_id' => 1,
    'date_time'    => array(
       'date' => 2,
       'time' => 3,
    ),
) );

### CODE REVIEW

File: gravity-forms/gw-update-posts.php [hunk 1]
```diff
@@ -92,6 +92,32 @@ public function update_post_by_entry( $entry, $form ) {
...
+				list( $hour, $min, $am_pm ) = array_pad( preg_split( '/[: ]/', $time ), 3, false );
+				if ( strtolower( $am_pm ) == 'pm' ) {
+					$hour += 12;
+				}
+			} else {
+				$hour = '00';
+				$min  = '00';
+			}
+
+			$new_date_time       = date( 'Y-m-d H:i:s', strtotime( sprintf( '%s %s:%s:00', $date, $hour, $min ) ) );
```
Comment (spivurno) [inline gravity-forms/gw-update-posts.php:116]: Ah, I know the other snippet used `date()` but our new standards do recommend using `gmdate()` instead.

File: gravity-forms/gw-update-posts.php [hunk 2]
```diff
@@ -92,6 +92,32 @@ public function update_post_by_entry( $entry, $form ) {
 			$post->post_name = rgar( $entry, $this->_args['slug'] );
 		}
 
+		if ( $this->_args['date_time'] ) {
+			$date = $this->_args['date_time']['date'];
+			$time = $this->_args['date_time']['time'];
+
+			$date = rgar( $entry, $date );
+			$time = rgar( $entry, $time );
```
Comment (spivurno) [inline gravity-forms/gw-update-posts.php:100]: I would recommend setting the a default time here like so:
 
`rgar( $entry, $time, '00:00 am' );`

File: gravity-forms/gw-update-posts.php [hunk 3]
```diff
@@ -92,6 +92,32 @@ public function update_post_by_entry( $entry, $form ) {
...
+			if ( empty( $date ) ) {
+				$date = explode( ' ', $post->post_date )[0];
+			}
+
+			if ( $time ) {
+				list( $hour, $min, $am_pm ) = array_pad( preg_split( '/[: ]/', $time ), 3, false );
+				if ( strtolower( $am_pm ) == 'pm' ) {
+					$hour += 12;
+				}
+			} else {
```
Comment (spivurno) [inline gravity-forms/gw-update-posts.php:111]: With the change above, we can ditch this `else` case.

**Review by spivurno — CHANGES_REQUESTED** [review]
Looking good! Requested a few tweaks.

**Comment by sbassah** [comment]
@spivurno I have made the suggested changes, please. I made a mistake with the commit message, I think I should have used tilde ~ for this. Please, is it possible to change the commit message?

**Review by spivurno — APPROVED** [review]
Perfect!

**Comment by spivurno** [comment]
@sbassah No worries on the commit message. I've forced merged this branch. 🙏

**Pull request merged**